middle_end/flambda2/tests/lib/minicheck/type.ml

```diff
@@ -12,259 +12,199 @@
 (*                                                                        *)
 (**************************************************************************)
 
-[@@@ocaml.warning "+a-4-30-40-41-42"]
+module Impl = struct
+  type 'a t =
+    { generator : 'a Generator.t;
+      shrinker : 'a Shrinker.t;
+      printer : 'a Printer.t
+    }
+end
 
-type 'a t =
-  { generate : Splittable_random.t -> 'a;
-    print : Format.formatter -> 'a -> unit
+type ('a, 'repr) t =
+  { impl : 'repr Impl.t;
+    get_value : 'repr -> 'a
   }
```

Comment:
I gather the idea here is that you want to allow generating a `'repr`, but working with an `'a`.  AFAICT, this is only necessary to support various kinds of composition?  This seems like a lot of extra complexity, and I'm wondering if it's a sign that this bundles too much together.